Sds.Osdr.EndToEndTests/Tests/Categories/DeleteEntityCategory.cs

```diff
@@ -0,0 +1,109 @@
+                {
+                    new TreeNode("My Test Category"),
+                    new TreeNode("Projects Two")
+                })
+            };
+
+            var response = harness.JohnApi.PostData("api/categorytrees/tree", categories).Result;
+
+            var content = response.Content.ReadAsStringAsync().Result;
+
+            RootCategoryId = Guid.Parse(content);
+
+            harness.WaitWhileCategoryTreePersisted(RootCategoryId);
+
+            BlobId = harness.JohnBlobStorageClient.AddResource(harness.JohnId.ToString(), "Chemical-diagram.png", new Dictionary<string, object>() { { "parentId", harness.JohnId } }).Result;
+
+            FileId = harness.WaitWhileFileProcessed(BlobId);
+        }
+    }
+
+    [Collection("OSDR Test Harness")]
+    public class DeleteEntityCategory : OsdrWebTest, IClassFixture<DeleteEntityCategoryFixture>
+    {
+        private Guid RootCategoryId;
+        public Guid BlobId { get; set; }
+        public Guid FileId { get; set; }
+
+
+        public DeleteEntityCategory(OsdrTestHarness harness, ITestOutputHelper output, DeleteEntityCategoryFixture fixture) : base(harness, output)
+        {
+            RootCategoryId = fixture.RootCategoryId;
+            BlobId = fixture.BlobId;
+            FileId = fixture.FileId;
+        }
+
+        [Fact, WebApiTrait(TraitGroup.All, TraitGroup.Categories)]
+        public async Task DeleteOneCategoryToEntity()
+        {
+            var fileNodeResponse = await JohnApi.GetNodeById(FileId);
+            var fileNode = JsonConvert.DeserializeObject<JObject>(await fileNodeResponse.Content.ReadAsStringAsync());
+            var fileNodeId = Guid.Parse(fileNode.Value<string>("id"));
+
+            var treeResponse = JohnApi.GetData($"api/categorytrees/tree/{RootCategoryId}").Result;
+            var treeContent = treeResponse.Content.ReadAsStringAsync().Result;
+            var categoryId1 = JObject.Parse(treeContent)["nodes"][0]["children"][0]["id"].ToString();
+            var categoryId2 = JObject.Parse(treeContent)["nodes"][0]["children"][1]["id"].ToString();
+
+            // add categories to entity
+            await JohnApi.PostData($"/api/categoryentities/entities/{fileNodeId}/categories", new List<string> { categoryId1, categoryId2 });
+            // check if node exists by categoryId1
+            var firstCategoryAddedNode = GetNodeByCategoryId(categoryId1);
+            firstCategoryAddedNode.Value<string>("id").Should().Be(fileNodeId.ToString());
+
+            // delete first category from node
+            await JohnApi.DeleteData($"/api/categoryentities/entities/{fileNodeId}/categories/{categoryId1}");
+            // check if node contains categoryId
+            var firstCategoryDeletedNode = GetNodeByCategoryId(categoryId1, false);
+            firstCategoryDeletedNode.Should().BeNull();
+        }
+
+        private JObject GetNodeByCategoryId(string categoryId, bool ifExists = true)
```

Comment:
you don't need `bool ifExists = true` instead you have to make direct query to ElasticSearch and make sure that record doesn't exist